### PR TITLE
`Development`: Add typing to `detail-list` component

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context
In order to comply with the design guidelines, this PR adds typing for the detail-list-component. Due to the needed discriminated union type, this was quite complex.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to some detail pages (lecture, course, exam, tutorial group, exercise)
3. Verify that the lists are still displayed.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [course-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6016/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/manage/detail/course-detail.component.ts.html) | 90.4% | ✅  |
| [tutorial-group-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6016/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/tutorial-groups/shared/tutorial-group-detail/tutorial-group-detail.component.ts.html) | 90.24% | ✅ |
| [detail-overview-list.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6016/latest/artifact/shared/Coverage-Report-Client-Tests/app/detail-overview-list/detail-overview-list.component.ts.html) | 98.03% | ✅ |
| [file-upload-exercise-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6016/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/file-upload/manage/file-upload-exercise-detail.component.ts.html) | 97.91% | ✅|
| [modeling-exercise-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6016/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/modeling/manage/modeling-exercise-detail.component.ts.html) | 93.05% | ✅ |
| [programming-exercise-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6016/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/programming/manage/programming-exercise-detail.component.ts.html) | 87.02% | ✅|
| [quiz-exercise-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6016/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/quiz/manage/quiz-exercise-detail.component.ts.html) | 100% | ✅ |
| [utils.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6016/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/shared/utils.ts.html) | 94.11% | ✅|
| [text-exercise-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6016/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/manage/text-exercise/text-exercise-detail.component.ts.html) | 100% | ✅ |

## Summary by CodeRabbit

- **New Features**
	- Enhanced detail overview display with improved conditional rendering and structure adjustment for various detail types.
- **Bug Fixes**
	- Fixed issues with router link assignment in course and tutorial group details for better navigation control.
	- Resolved detail rendering issues by refining conditions and removing unnecessary filter operations across multiple components.
- **Refactor**
	- Streamlined detail-related imports and adjusted logic for setting `routerLink` values and detail array construction.
	- Reorganized and improved functions for downloading diagrams and showing git diffs to handle optional parameters more effectively.
- **Tests**
	- Updated component tests for detail overview list and quiz exercise details, including import adjustments and simplification of test logic.

### CODE REVIEW

**Review by georgyia — APPROVED** [review]
manually tested on ts2, works as expected

**Review by reschandreas — APPROVED** [review]
tested on TS4, works as expected

**Review by rstief — APPROVED** [review]
Code looks good 👍

File: src/main/webapp/app/detail-overview-list/detail-overview-list.component.html [hunk 1]
```diff
@@ -5,264 +5,282 @@
...
-                                    {{ detail.data.text }}
-                                </a>
+                @switch (detail.type) {
+                    @case (DetailType.Text) {
+                        <dd id="detail-value-{{ detail.title }}">
+                            @if (detail.data.text) {
+                                {{ detail.data.text }}
                             } @else {
-                                <span>{{ detail.data.text }}</span>
+                                <ng-container *ngTemplateOutlet="noData"></ng-container>
```
Comment (MaximilianAnzinger) [inline src/main/webapp/app/detail-overview-list/detail-overview-list.component.html:23]:
```suggestion
                                <ng-container *ngTemplateOutlet="noData" />
```

**Review by MaximilianAnzinger — CHANGES_REQUESTED** [review]

**Review by rstief — APPROVED** [review]
reapprove

**Review by laurenzfb — APPROVED** [review]

**Review by MaximilianAnzinger — APPROVED** [review]

**Pull request merged**